### PR TITLE
Feature: Convolve1D with short and long filters

### PR DESCRIPTION
**Pull request by mrava87**
This PR generalizes the `Convolve1D` operator to handle both the scenario of short filter (and longer input) as well as short input (and longer filter). The latter is introduced here.

### CODE REVIEW

**Comment by mrava87** [comment]
@cako when you have time, would be good to get some feedback before merging :)

**Comment by cako** [comment]
I will have a look tomorrow!

File: pylops/signalprocessing/convolve1d.py [hunk 1]
```diff
@@ -34,28 +39,187 @@ def _choose_convfunc(
...
+        self.h = h
+        self.offset = 2 * (self.nh // 2 - int(offset))
+        if self.nh % 2 == 0:
+            self.offset -= 1
+        if self.offset != 0:
+            self.h = _pad_along_axis(
+                self.h,
+                (
+                    self.offset if self.offset > 0 else 0,
+                    -self.offset if self.offset < 0 else 0,
```
Comment (cako) [inline pylops/signalprocessing/convolve1d.py:85]: `max(self.offset, 0), -min(self.offset, 0)`?

**Review by cako — APPROVED** [review]
Looks good to me!

File: pylops/signalprocessing/convolve1d.py [hunk 2]
```diff
@@ -34,28 +39,187 @@ def _choose_convfunc(
...
+            self.offset -= 1
+        self.hstar = np.flip(self.h, axis=-1)
+
+        self.pad = np.zeros((len(dims), 2), dtype=int)
+        self.pad[self.axis, 0] = self.offset if self.offset > 0 else 0
+        self.pad[self.axis, 1] = -self.offset if self.offset < 0 else 0
+
+        self.padd = np.zeros((len(dims), 2), dtype=int)
+        self.padd[self.axis, 1] = self.offset if self.offset > 0 else 0
+        self.padd[self.axis, 0] = -self.offset if self.offset < 0 else 0
```
Comment (cako) [inline pylops/signalprocessing/convolve1d.py:154]: min/max offset here too

**Pull request merged**